### PR TITLE
Support async access token scope validators

### PR DESCRIPTION
**Pull request by drtyh2o**
Small change to support asynchronous as well as synchronous scope validation.

Also, a few dependency updates and a couple of ESLint/Prettier fixes.

The only notable code change is [here](https://github.com/lacuna-tech/mds-core/pull/147/files#diff-53d96eb72707cd2764caae5164b65005)

### CODE REVIEW

File: packages/mds-api-server/index.ts [hunk 1]
```diff
@@ -131,14 +131,16 @@ export const ApiServer = (
...
-  res: ApiResponse,
-  next: express.NextFunction
-) => {
-  if (process.env.VERIFY_ACCESS_TOKEN_SCOPE === 'false' || validator(res.locals.scopes)) {
-    next()
-  } else {
-    res.status(403).send({ error: new AuthorizationError('no access without scope', { claims: res.locals.claims }) })
-  }
-}
+export const checkAccess = (validator: (scopes: AccessTokenScope[]) => boolean | Promise<boolean>) =>
```
Comment (karcass) [inline packages/mds-api-server/index.ts]: Coming back up to speed on my TS.  It looks like both return paths are async -- doesn't that imply that the return value is always `Promise<boolean>`?

File: packages/mds-policy-author/request-handlers.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import log from '@mds-core/mds-logger'
 import db from '@mds-core/mds-db'
-import { PolicyApiRequest, PolicyApiResponse } from './types'
 import { BadParamsError } from '@mds-core/mds-utils'
+import { PolicyApiRequest, PolicyApiResponse } from './types'
 
-const getPolicies = async (req : PolicyApiRequest, res : PolicyApiResponse) => {
+const getPolicies = async (req: PolicyApiRequest, res: PolicyApiResponse) => {
```
Comment (karcass) [inline packages/mds-policy-author/request-handlers.ts]: should this have a type for the return value?

Comment (drtyh2o) [reply packages/mds-policy-author/request-handlers.ts]: Maybe, was just a linter/format change in this PR.

File: packages/mds-api-server/index.ts (hunk 1 above)
Comment (drtyh2o) [reply packages/mds-api-server/index.ts]: `checkAccess` itself actually returns an async middleware `(req, res, next)` function. 

The async middleware function is capable of using either a sync or async `validator` function, hence the `boolean | Promise<boolean>`

So, while the middleware is typically something simple like this:
```ts
checkAccess(scopes => scopes.includes('admin:all'))
```
The middleware could also use some more complex async validation (this is a contrived example for illustration)
```ts
checkAccess(async scopes => {
  // do some async validation
  const valid = await Promise.resolve(scopes.includes('admin:all'))
  return valid
})
```

**Pull request merged**